### DISCUSSION TITLE
【Region China】 2.7.0 and 2.9.0 change login validation twice.

### DISCUSSION THREAD

**Discussion by HuChundong** [bimmerconnected/bimmer_connected#488] [Show and tell]
all done.

**Comment by rikroe** [comment]
Very nice, congratulation!

Due to the nature of the library, using a captcha would be a lot of work (i.e. like you apparently did). Is the captcha occuring every time when logging in with username/password?

Also, would you share your work at some point?

**Reply by HuChundong** [reply]
> Very nice, congratulation!
> 
> Due to the nature of the library, using a captcha would be a lot of work (i.e. like you apparently did). Is the captcha occuring every time when logging in with username/password?
> 
> Also, would you share your work at some point?

1. captcha will show every time in 2.9.0. but it's easy to crack. bmw‘s captcha is a joke.
2. if someone crack it too and use it for business, i will share my work.  but at this time. 2.3's api still work.

**Reply by Bpazy** [reply]
@HuChundong 2.3's api outdated. How about sharing your work?
